kyo-core/shared/src/main/scala/kyo/ios.scala

```diff
@@ -89,7 +89,7 @@ sealed trait IOs extends Effect[IOs]:
         @tailrec def runLoop(v: T < IOs): T =
             v match
                 case kyo: Suspend[IO, Unit, T, IOs] @unchecked =>
-                    require(kyo.tag == tag, "Unhandled effect: " + kyo.tag)
+                    bug(kyo.tag != tag, "Unhandled effect: " + kyo.tag.parse)
```

Comment:
Perhaps it's useful to make a helper function for this specific case:
```
def ensureTagsEq(handlerTag: Tag, effectTag: Tag): Unit =
  if (handlerTag != effectTag) bug(...)
```

And ensure that both tags are printed. I'm guessing when users file a ticket, it would be useful to know which handler threw the exception.